src/plugins/data_source/README.md

```diff
@@ -0,0 +1,13 @@
+# data_source
```

Comment:
add a task to track populate this readme https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2053